src/java/org/apache/cassandra/io/util/FileUtils.java

```diff
@@ -68,19 +71,29 @@
     private static final boolean canCleanDirectBuffers;
     private static final AtomicReference<FSErrorHandler> fsErrorHandler = new AtomicReference<>();
 
+    private static Class clsDirectBuffer;
+    private static MethodHandle mhDirectBufferCleaner;
+    private static MethodHandle mhCleanerClean;
+
     static
     {
         boolean canClean = false;
         try
         {
+            clsDirectBuffer = Class.forName("sun.nio.ch.DirectBuffer");
```

Comment:
To clarify: on quip, you mention that direct memory allocation is only possible in JDK8, and not JDK 11. Did we confirm that is the case? If so, I think it may be worth exploring using sun.misc.Unsafe.invokeCleaner(ByteBuffer) below instead then (and purposefully disabling direct memory support for jdk8), as I think we'll see a major perf hit if we can no longer perform direct memory allocations.

Reply:
The doc needs an update. TL;DR; is that cleanup is best effort and depends on JPMS flags giving our code access to `sun.nio.ch.DirectBuffer`. Opening up access to the DirectBuffer is done via `JVM_OPTS="$JVM_OPTS --add-exports java.base/sun.nio.ch=ALL-UNNAMED"` so this now does work.

Reply:
Awesome. The only other thing then I'll flag (which is continuing on my comment below), is that we should probably ensure we never do direct memory allocation if we cannot cleanup. My understanding is that direct memory allocations will never be cleaned up otherwise, and we will eventually hit out-of-memory.

Reply:
The runtime-exception is propagated to the main thread if either reflection on internal apis isn't enable or if the `sun.nio.ch.DirectBuffer` is unavailable for some reason, this would prevent startup before any memory access takes place (at the static initialization stage). This is about as safe as it gets. 

FWIW, I though about capturing the exception and placing the node in disabled mode with a logging of non-transient error but this just doesn't make sense to do. If the node is configured to use direct memory buffers it should just be allowed to crash before it can create / acess new memtables as that would just simply fail. We should never end up in a scenario where the node is configured for direct buffers but isn't allowed to use reflection to verify that `sun.nio.ch.DirectBuffer` is available.

Reply:
I would also suggest we add `clean(buf)`, to ensure our actual invoke works as well, to be extra safe. Agreed on not making this a non-transient error.  

So:
```
            clsDirectBuffer = Class.forName("sun.nio.ch.DirectBuffer");
            Method mDirectBufferCleaner = clsDirectBuffer.getMethod("cleaner");
            mhDirectBufferCleaner = MethodHandles.lookup().unreflect(mDirectBufferCleaner);
            Method mCleanerClean = mDirectBufferCleaner.getReturnType().getMethod("clean");
            mhCleanerClean = MethodHandles.lookup().unreflect(mCleanerClean);

            ByteBuffer buf = ByteBuffer.allocateDirect(1);
            clean(buf);
            ...
```

Reply:
This is a good idea, pushed a commit to address forcing a call to FileUtils#clean during the initialization of the cleaner.